localstack-core/localstack/aws/spec.py

```diff
@@ -303,5 +309,61 @@ def save_service_index_cache(index: LazyServiceCatalogIndex, file_path: str) ->
         target_prefix_index=index.target_prefix_index,
     )
     with open(file_path, "wb") as fd:
-        pickle.dump(cache, fd)
+        # use dill (instead of plain pickle) to avoid issues when serializing the pickle from __main__
+        dill.dump(cache, fd)
     return cache
+
+
+def _get_catalog_filename():
+    ls_ver = VERSION.replace(".", "_")
+    botocore_ver = botocore.__version__.replace(".", "_")
+    return f"service-catalog-{ls_ver}-{botocore_ver}.dill"
+
+
+@singleton_factory
+def get_service_catalog() -> ServiceCatalog:
+    """Loads the ServiceCatalog (which contains all the service specs), and potentially re-uses a cached index."""
+
+    try:
+        catalog_file_name = _get_catalog_filename()
+        static_catalog_file = os.path.join(config.dirs.static_libs, catalog_file_name)
+
+        # try to load or load/build/save the service catalog index from the static libs
+        index = None
+        if os.path.exists(static_catalog_file):
```

Comment:
If the current localstack version or botocore version is different, then the static cache should be invalid, no?

This could happen when mounting a different localstack version into the container with the container dev mode I think

Reply:
nvm, it would look up a different directory then because the filename is generated based on the current version. was confused at first

Reply:
Also the filename changed ;)